bin/collator/src/parachain/service.rs

```diff
@@ -1202,255 +1158,10 @@ pub async fn start_astar_node(
-                        block_proposal_slot_portion: SlotProportion::new(1f32 / 24f32),
-                        // And a maximum of 750ms if slots are skipped
-                        max_block_proposal_slot_portion: Some(SlotProportion::new(1f32 / 16f32)),
-                        telemetry: telemetry2,
-                    })
-                }),
-            ));
-
-            let mut proposer_factory =
-                sc_basic_authorship::ProposerFactory::with_proof_recording(
-                    task_manager.spawn_handle(),
-                    client.clone(),
-                    transaction_pool,
-                    prometheus_registry,
-                    telemetry.clone(),
-                );
-
-            proposer_factory.set_default_block_size_limit(additional_config.proposer_block_size_limit);
-            proposer_factory.set_soft_deadline(Percent::from_percent(additional_config.proposer_soft_deadline_percent));
-
-            let relay_chain_consensus =
-                cumulus_client_consensus_relay_chain::build_relay_chain_consensus(
-                    cumulus_client_consensus_relay_chain::BuildRelayChainConsensusParams {
-                        para_id: id,
-                        proposer_factory,
-                        block_import: block_import, //client.clone(),
-                        relay_chain_interface: relay_chain_interface.clone(),
-                        create_inherent_data_providers:
-                            move |_, (relay_parent, validation_data)| {
-                                let relay_chain_for_aura = relay_chain_interface.clone();
-                                async move {
-                                    let parachain_inherent =
-                                        cumulus_primitives_parachain_inherent::ParachainInherentData::create_at(
-                                            relay_parent,
-                                            &relay_chain_for_aura,
-                                            &validation_data,
-                                            id,
-                                        ).await;
-                                    let parachain_inherent =
-                                        parachain_inherent.ok_or_else(|| {
-                                            Box::<dyn std::error::Error + Send + Sync>::from(
-                                                "Failed to create parachain inherent",
-                                            )
-                                        })?;
-                                    Ok(parachain_inherent)
-                                }
-                            },
-                    },
-                );
-
-            let parachain_consensus = Box::new(WaitForAuraConsensus {
-                client,
-                aura_consensus: Arc::new(Mutex::new(aura_consensus)),
-                relay_chain_consensus: Arc::new(Mutex::new(relay_chain_consensus)),
-            });
```

Comment:
Sorry if I'm missing it, but I don't see where this part has been moved? Won't it break Shiden sync?

Reply:
the block import isn't changed. this is block authoring and code is here https://github.com/AstarNetwork/Astar/pull/1263/files/daf8a547131e08f891edf9c234473ac0d1ab28c3#diff-e7ed72ccf74167731e3cfe0770979224242dfd278d03768448761ad634cac229R573-R588

Reply:
Oh, I see, the requirement imposed by the runtime API check was actually redundant so this `WaitForAuraConsensus` is no longer needed.

Is it then safe to say that the `shell_upgrade` mod can be completely deleted?

Reply:
it's still used by `build_import_queue` for shiden

Reply:
you're right `WaitForAuraConsensus` actually not needed but `shell_upgrade` is still needed